private_gpt/components/nlsql/nlsql_component.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from urllib.parse import quote_plus
+
+from injector import inject, singleton
+from llama_index import ServiceContext, SQLDatabase, VectorStoreIndex
+from llama_index.indices.struct_store import SQLTableRetrieverQueryEngine
+from llama_index.objects import ObjectIndex, SQLTableNodeMapping, SQLTableSchema
+from sqlalchemy import (
+    MetaData,
+)
+from sqlalchemy.engine import create_engine
+from sqlalchemy.engine.base import Engine
```

Comment:
Did you try without the `local` mode?

Are these dependencies shipped from `privateGPT` by default?

It looks to me that these additional dependencies are brought by default, while it should be an `opt-in` feature I think. This would also means that this `NLSQLComponent` should be conditionally injected in the app context (basically, the app should still be able to run without that SQL component).

Ideally, the app should load this component only if the configuration is specifying a valid SQL configuration (thanks to some "on/off" switch in configuration maybe)